packages/nimble-components/src/table-column/text/template.ts

```diff
@@ -7,6 +7,16 @@ TableCellState<TableColumnTextCellRecord, TableColumnTextColumnConfig>
 >`
     <span
         class="${x => (typeof x.cellRecord.value === 'string' ? '' : 'placeholder')}"
+        @mouseover="${(_x, c) => {
+        const span = c.event.target as HTMLSpanElement;
+        if (
+            span?.textContent?.trim()
+                && span.offsetWidth < span.scrollWidth
+        ) {
+            span.setAttribute('title', span.textContent.trim());
+        }
```

Comment:
Hmm we essentially are showing that we need instance state that the plugin authors can manipulate. Here this is requiring that mouse over and mouse out are called reliably back to back before any other state changes. That's not a strong guarantee.

Many different state changes could happen between mouseover and mouseout:

- the placeholder could change
- the cellRecord.value could change
- the data could change

Any of the above would mean make the state of title incorrect.

What we want is to be able to model this as state changes like we would do anywhere elsewhere:
```ts
class MyComponent {
    @observable isHovered;
    @observable cellRecord;
    @observable columnConfig;
    mySpan;

    @volatile
    content () {
         return typeof this.cellRecord.value === 'string'
        ? this.cellRecord.value
        : this.columnConfig.placeholder
    }
}

const template = html<MyComponent >`
    <span ${ref('mySpan')}
        @mouseover="${x => { /* do some logic and */ x.isHovered = true}}"
        @mouseout="${x => {x.isHovered = false}}"
        title=${x=>x.isHovered ? x.content() : null}
    >${x => x.content()}</span>
`
```

The above is more in-line with how template logic is expected to be written, doesn't suffer from state tracking issues, and is more maintainable by avoiding mixing template logic and imperative dom manipulation.

Reply:
While all of the above is true, I think the discussion for it would continue at least right now in the cell state scrolling hld: https://github.com/ni/nimble/pull/1052#discussion_r1116358731

I don't think the title state going bad is critical enough to block this PR. 

For this PR can you just move this logic outside the template to keep the template easier to read:
```ts
const setTitleWhenOverflow = (span) => { /* do stuff */};
const template = html`
 <span @mouseover="${(_x, c) => { setTitleWhenOverflow(c.event.target) }"></span>
`
```

Reply:
Extracted the function to improve readability of the template.